src/Microsoft.AspNetCore.Mvc.Razor/Internal/FileProviderRazorProjectItem.cs

```diff
@@ -27,11 +30,50 @@ public FileProviderRazorProjectItem(IFileInfo fileInfo, string basePath, string
 
         public override string PhysicalPath => FileInfo.PhysicalPath;
 
-        public override string RelativePhysicalPath { get; }
+        public override string RelativePhysicalPath
+        {
+            get
+            {
+                if (_relativePhysicalPath == null)
+                {
+                    if (Exists)
+                    {
+                        if (_root != null && !string.IsNullOrEmpty(PhysicalPath) && PhysicalPath.StartsWith(_root))
+                        {
+                            _relativePhysicalPath = PhysicalPath.Substring(_root.Length + 1); // Include leading separator
+                        }
+                        else
+                        {
+                            // Use FilePath if the file is not directly accessible
+                            _relativePhysicalPath = NormalizeAndEnsureValidPhysicalPath(FilePath);
+                        }
```

Comment:
I'm not sure about this *else*. FilePath and RelativePhysicalPath are not meanty to be interchangable.

Reply:
We likely don't need this here. Calling code can fallback to `FilePath` if this value is null (which we do now in Razor). In scenarios like embedded file provider, this will be null, but I there wouldn't be any design-time scenarios that are affected by this.

Reply:
Yup. It should be safe to remove this now after @pranavkm's change in Razor